### ISSUE TITLE
Kafka scaler: excludePersistentLag not working for partitions with invalid offset (that is -1)

### ISSUE THREAD

**Issue by rakesh-ism** [kedacore/keda#5274]
### Report

When I set excludePersistentLag flag to true, it does not excludes persistent lags for the partition with invalid offset (that is -1). 

### Expected Behavior

If the lags for these partitions with invalid offset is persistent, it should be ignored in the custom metrics.

### Actual Behavior

In my test set up, this resulted in no of kafka consumer pods to set to max.

### Steps to Reproduce the Problem

1. In the setup, you should have some partitions that has not been consumed till now.  (Invalid offset)
2. There should be no new message coming to the partition.
3. Old messages are not in kafka buffer as they are already expired.

Use kafka scalar to scale the consumer deployment.

I know this is might not be a real production set up. I had this set up in my sandbox landscape for . And it took me hard time to find out what was the problem.

### Logs from KEDA operator

```
example
```

### KEDA Version

2.12.1

### Kubernetes Version

None

### Platform

Other

### Scaler Details

Kafka

**Comment by zroubalik** [comment]
@dttung2905 What do you think about this?

**Comment by dttung2905** [comment]
@zroubalik I think its a good point that @rakesh-ism bring up and imo we can implement the logic. I do agree that sometimes we do get negative offset lag for kafka consumer (quite common from my experience managing kafka actually 😞 ) I will create a PR tomorrow morning for this ( its late in my timezone now haha )

**Comment by dttung2905** [comment]
@rakesh-ism could you help to send over a few things ?
- your kafka scaler config that you use?
- logs from the keda operator
From my quick look at the code, if your consumer offset for a partition is -1, it should be caught within this code block already 🤔 
https://github.com/kedacore/keda/blob/923583f0477e44ae5426777925401820ec0ad6ba/pkg/scalers/kafka_scaler.go#L667-L686

**Comment by rakesh-ism** [comment]
@dttung2905 
Thanks for taking it up.
Attached is the keda config I used. As the 
[keda.txt](https://github.com/kedacore/keda/files/13815492/keda.txt)

In my case, offsetresetpolicy is "earliest", the above code is bypassed. and code branch at line no 684 is executed.

**Comment by dttung2905** [comment]
Thanks for sharing with us the config file.
Hmm this is weird ! Could you share any long from the controller too ? If you can share the steps / setup to replicate the problem that would be great for us :pray:

**Comment by dttung2905** [comment]
@rakesh-ism sure. Let me know if you manage to reproduce it :pray:

**Comment by rakesh-ism** [comment]
The issue got reproduced in my system again. I did some analysis and found that we have a topic in the consumer group where the lag is like below. Can you please check?

|GROUP |TOPIC |PARTITION |CURRENT-OFFSET |LOG-END-OFFSET|LAG|CONSUMER-ID|HOST|CLIENT-ID|
|--------|--------|--------|--------|--------|--------|--------|--------|--------|
|group1|topic1|0|80|80|0|rdkafka-xxx-xxx-xxx-xxx-xxx|/xx.xx.xx.xx|rdkafka|
|group1 |topic1|1|-|60|-|rdkafka-xxx-xxx-xxx-xxx-xxy|/100.100.17.69|rdkafka|
|group1|topic1|2|-|96|-|rdkafka-xxx-xxx-xxx-xxx-xxz|/100.100.14.23 |rdkafka|

**Comment by zroubalik** [comment]
@dttung2905 PTAL 🙏 :)

**Comment by somabodonyi** [comment]
@rakesh-ism @zroubalik

This fix for this issue has not been released yet.
The issue was fixed in May, but the latest release (2.14.0) was in April.

**Comment by rakesh-ism** [comment]
Any updates

**Comment by danielmotaleite** [comment]
TLDR: update at least to keda 2.15.x,  enable scaleToZeroOnInvalidOffset: "true" and add a cron trigger to start the consumer at least one a day/hour (depending of your requirements). Long term fix below, by forcing one message in all partitions.



I also have this problem and found that this option fix this:

scaleToZeroOnInvalidOffset: "true"

So this option will consider lag zero when the lag is below zero or non-existent, fixing this problem.
This may cause other problem, like if you scale to zero and a message arrives to a partition without lag, as there is no consumer, it will not be consumed nor detected (as there is no consumer lag yet). If a new message arrives to the partition that already have a lag metric, everything works correctly, so the danger is only on the partitions without lag.

 So this option is safe is your minimum consumer is at least one, but if scale to zero, you may need to start the consumer (maybe via cron trigger) to force the consumer to start every few hours/days to make sure you don't have any stale message in the partitions without lag.
 
 when all partitions already have lag metric, you can disable this option.
 So a better long run solution maybe probably use kcat to force produce a message to all partitions, to force the consumer to work at least once in all partitions. Of course, either prepare your consumer for a (invalid) test message, or make sure you inject a valid messaged, or else you may lock your consumer with a broken message that never gets consumed

Kafka lag negative is usually a quick corner case of a active consumer, it is usually now a major issue and the default  scaleToZeroOnInvalidOffset: "false" already takes care of this. Please make sure you use the latest keda version, as older versions may not have this option and the version 2.15.0 add a fix for when the  offsetResetPolicy: "earliest" not applying the  scaleToZeroOnInvalidOffset workaround.

So for my side, i updated keda, enabled that option and it solved my issue. i them enabled cron to start the consumer once a day, just in case for the topics with very little activity.
i'm also  waiting for the devs to add a discard processing when the message is of type { "type": "debug" } (payload dependent of your consumer) to manually initialize all partitions
